src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs

```diff
@@ -108,6 +116,13 @@ public ObjectResult(object value)
             IOutputFormatter selectedFormatter = null;
             if (ContentTypes == null || ContentTypes.Count == 0)
             {
+                var requestContentType = formatterContext.ActionContext.HttpContext.Request.ContentType;
+
+                if (sortedAcceptHeaderMediaTypes.Count() == 0 && requestContentType == null)
+                {
+                    return SelectFormatterBasedOnTypeMatch(formatterContext, formatters);
```

Comment:
Add a comment here - "there's nothing in the request that we can use to select a content type so..."